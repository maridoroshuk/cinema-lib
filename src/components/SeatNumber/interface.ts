export interface ISeatProps {
  children: string;
  backgroundColor: string;
  index: number;
}
export interface ISeatNumberProps {
  backgroundColor: string;
  seatIndex: number;
}
