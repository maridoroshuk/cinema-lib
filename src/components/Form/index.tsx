import React from 'react';
import { Formik } from 'formik';

import { IFormProps } from './interface';
import { StyledForm } from './styles';

export const Form = <T extends {}>({
  children,
  initialValues,
  validate,
  onFormSubmit,
}: IFormProps<T>) => {
  return (
    <Formik
      initialValues={initialValues}
      validate={validate}
      onSubmit={onFormSubmit}
    >
      <StyledForm>{children}</StyledForm>
    </Formik>
  );
};
