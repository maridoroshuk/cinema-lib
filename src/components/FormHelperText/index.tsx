import React from 'react';

import { IFormHelperTextProps } from './interface';
import { Wrapper } from './styles';

export const FormHelperText = ({ message }: IFormHelperTextProps) => {
  return <Wrapper>{message}</Wrapper>;
};
