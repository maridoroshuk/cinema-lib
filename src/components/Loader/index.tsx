import React from 'react';
import ReactLoading from 'react-loading';
import { ILoaderProps } from './interface';
import { LoaderWrapper } from './styles';

export function Loader({ height, width, color }: ILoaderProps) {
  return (
    <LoaderWrapper>
      <ReactLoading type="spokes" color={color} height={height} width={width} />
    </LoaderWrapper>
  );
}
