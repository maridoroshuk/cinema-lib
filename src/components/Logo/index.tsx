import React from 'react';

import LogoImage from '@/assets/Logo.png';
import { getLogoSize } from '@/helpers/getLogoSize';

import { LogoStyled } from './styles';
import { ILogo } from './interface';

export const Logo = ({ size }: ILogo) => {
  const width = getLogoSize(size);
  return (
    <LogoStyled src={LogoImage} width={width} alt="company logo (Modsen)" />
  );
};
