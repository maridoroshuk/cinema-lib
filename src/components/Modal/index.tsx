import React from 'react';

import { IModal } from './interface';
import { Wrapper } from './styles';

export const Modal = ({ children }: IModal) => {
  return <Wrapper>{children}</Wrapper>;
};
