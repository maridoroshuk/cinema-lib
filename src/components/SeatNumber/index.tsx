import React from 'react';

import { ISeatProps } from '@/components/SeatNumber/interface';
import { StyledSeat } from '@/components/SeatNumber/styles';

export const SeatNumber = ({
  children,
  backgroundColor,
  index,
}: Partial<ISeatProps>) => {
  return (
    <StyledSeat backgroundColor={backgroundColor} seatIndex={index}>
      {children}
    </StyledSeat>
  );
};
