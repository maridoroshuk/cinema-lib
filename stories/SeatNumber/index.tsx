import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import { SeatNumber } from '@/components/SeatNumber';

export default {
  title: 'Buttons/Button',
  component: SeatNumber,
  argTypes: {
    backgroundColor: {
      control: 'radio',
      options: ['gold', 'neutral050'],
    },
  },
} as ComponentMeta<typeof SeatNumber>;

const Template: ComponentStory<typeof SeatNumber> = (args) => (
  <SeatNumber {...args} />
);
export const Primary = Template.bind({});
Primary.args = {
  children: 'SeatNumber',
  backgroundColor: 'neutral050',
};
export const Secondary = Template.bind({});
Secondary.args = {
  children: 'SeatNumber',
  backgroundColor: 'neutral050',
};
export const Last = Template.bind({});
Last.args = {
  backgroundColor: 'neutral050',
  children: 'SeatNumber',
};

export const Previous = Template.bind({});
Previous.args = {
  backgroundColor: 'neutral050',
  children: 'SeatNumber',
};

export const Current = Template.bind({});
Current.args = {
  backgroundColor: 'gold',
  children: 'SeatNumber',
};
