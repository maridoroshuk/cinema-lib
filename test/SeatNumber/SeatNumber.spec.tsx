import React from 'react';
import { render } from '@testing-library/react';

import { SeatNumber } from '@/components/SeatNumber';

import 'jest-styled-components';

import { setupTestProvider } from '../setupTests';

describe('SeatNumber', () => {
  it('renders', () => {
    const { container } = render(
      setupTestProvider(
        <SeatNumber backgroundColor="gold" index={1}>
          1
        </SeatNumber>,
      ),
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
