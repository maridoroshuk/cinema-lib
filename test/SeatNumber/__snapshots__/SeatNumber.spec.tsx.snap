// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeatNumber renders 1`] = `
.c0 {
  border-radius: 10px;
  border: 1px solid #000000;
  font-weight: 300;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #D98639;
  color: #000000;
  width: 115px;
  height: 110px;
}

<div
  class="c0"
>
  1
</div>
`;
